src/main/java/com/weaverplatform/import/xmi/XmiImporter.java

```
@@ -0,0 +1,270 @@
+
+          List<XML> associationEnds = nodeAssociation.nodes(xpath);
+
+          for(XML associationEnd: associationEnds){
+
+            String type = xmiImporter.getValue(xmiImporter.getNamedItemOnNode(associationEnd, "type"));
+
+            xpath = "//UML.ModelElement.taggedValue/UML.TaggedValue";
+            List<XML> taggedValues = associationEnd.nodes(xpath);
+
+            for(XML taggedValue : taggedValues){
+              String tValue = xmiImporter.getValue(xmiImporter.getNamedItemOnNode(taggedValue, "value"));
+              if(tValue.equals("source")){
+
+                String oriParentName = aIndividualMap.get(type);
+
+                HashMap<String, Object> attributes = new HashMap<>();
+                attributes.put("label", associationName);
+                attributes.put("celltype", "individual");
+                xmiImporter.toWeaverAnnotation(attributes, oriParentName);
+
+              }
+            }
+
+          }
+
+        }
+      }
+
+
+    }
+
+  }
+
+  /**
+   * Reads an xmi file from a classpath (test resource directory) or unixpath
+   * @return InputStream from contents
+   */
+  public InputStream read(){
+
+    try{
+
+      if(!USE_UNIX_FILEPATH) {
+
+        byte[] contents = FileUtils.readFileToByteArray(new File(getClass().getClassLoader().getResource(FILE).getFile()));
+        InputStream in = new ByteArrayInputStream(contents);
+        InputStream cont = new ByteArrayInputStream(IOUtils.toByteArray(in));
+        return cont;
+
+      }
+
+      File f = new File(FILE);
+
+      boolean isFile = f.exists();
+
+      if(isFile){
+
+        Path path = Paths.get(f.getAbsolutePath());
+        byte[] data = Files.readAllBytes(path);
+
+        InputStream in = new ByteArrayInputStream(data);
+        InputStream cont = new ByteArrayInputStream(IOUtils.toByteArray(in));
+        return cont;
+      }
+
+    }catch(Exception e) {
+      System.out.println("cannot read!");
```

Catch more specific exceptions, do not eat exceptions in general. Maybe wrap the exception in a RuntimeException so the application will exit, as proceeding when an exception is thrown here seems pretty useless.